src/firefly/js/ui/DownloadDialog.jsx

```diff
@@ -161,6 +162,12 @@ export function DownloadOptionPanel (props) {
             return showInfoPopup('You have not chosen any data to download', 'No Data Selected');
         }
 
+        const {valid, message} = validateOnSubmit?.(formInputs) ?? {valid : true, message: ''};
```

Comment:
You might want a default message. Also when valid it true you don’t need `message`. So…
```js

const {valid, message=‘Invalid Input’} = validateOnSubmit?.(formInputs) ?? {valid : true};
```

Reply:
OK, I've added default message in if block instead.